tcpproxy.go

```diff
@@ -0,0 +1,104 @@
+// Copyright 2015 Canonical Ltd.
+// Licensed under the LGPLv3, see LICENCE file for details.
+
+package testing
+
+import (
+	"io"
+	"net"
+	"sync"
+
+	jc "github.com/juju/testing/checkers"
+	gc "gopkg.in/check.v1"
+)
+
+// TCPProxy is a simple TCP proxy that can be used
```

Comment:
Perhaps the name should be changed to emphasise its function, which is to cause connections to fail.